## Import Pandas and Load File

In [2]:
import pandas as pd

In [24]:
df = pd.read_csv("olympics_1896_2004.csv")

In [25]:
df.head(5)

This is a list of medallists at the Summer Olympic Games (1896-2004) with details of the year of the olympics, city, sport, discipline, athlete name, gender, event and medal  \
0                                                NaN                                                                                                                              
1  DISCLAIMER: No guarantees, express or implied,...                                                                                                                              
2                                                NaN                                                                                                                              
3                                                NaN                                                                                                                              
4                                               Year                                                                                                                              

  Unnamed: 1 Unnamed: 2  Unnamed: 3    Unnamed: 4 Unnamed: 5 Unnamed: 6  \
0        NaN        NaN         NaN           NaN        NaN        NaN   
1        NaN        NaN         NaN           NaN        NaN        NaN   
2        NaN        NaN         NaN           NaN        NaN        NaN   
3        NaN        NaN         NaN           NaN        NaN        NaN   
4       City      Sport  Discipline  Athlete Name        NOC     Gender   

  Unnamed: 7    Unnamed: 8 Unnamed: 9 Unnamed: 10  
0        NaN           NaN        NaN         NaN  
1        NaN           NaN        NaN         NaN  
2        NaN           NaN        NaN         NaN  
3        NaN           NaN        NaN         NaN  
4      Event  Event Gender      Medal    Position

In [59]:
# pd.read_csv?

In [27]:
df = pd.read_csv("olympics_1896_2004.csv", skiprows=5)
df.head(2)

Year    City    Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens  Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens  Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   

   Event Event Gender   Medal  Position  
0  100km            M    Gold         1  
1  100km            M  Silver         2

In [61]:
# dir(df)

## Observations

In [42]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27174 entries, 0 to 27173
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Year          27174 non-null  int64 
 1   City          27174 non-null  object
 2   Sport         27174 non-null  object
 3   Discipline    27174 non-null  object
 4   Athlete Name  27174 non-null  object
 5   NOC           27174 non-null  object
 6   Gender        27174 non-null  object
 7   Event         27174 non-null  object
 8   Event Gender  27174 non-null  object
 9   Medal         27174 non-null  object
 10  Position      27174 non-null  int64 
dtypes: int64(2), object(9)
memory usage: 2.3+ MB
None


In [51]:
df.rename(columns={"Athlete Name":"Athlete_Name","Event Gender":"Event_Gender"}, inplace=True)

### How many male and female atheletes were participated in Olympic 1896 ?

In [56]:
df["Gender"].value_counts()

Gender
Men      20611
Women     6563
Name: count, dtype: int64

### How many GOLD🏅 won by male and female?

In [81]:
df[df["Medal"] == "Gold"]['Gender'].value_counts()

Gender
Men      7003
Women    2178
Name: count, dtype: int64

### Which top 3 cities won most Bronze🥉?

In [86]:
df[df["Medal"] == "Bronze"]["City"].value_counts().head(3)

City
Athens         719
Los Angeles    696
Sydney         685
Name: count, dtype: int64

### How many Silver🥈 won by male in top 3 NOC?

In [89]:
df[(df["Medal"] == "Silver") | (df["Gender"] == "Men")]["NOC"].value_counts().head(3)

NOC
USA    3248
URS    1621
GBR    1377
Name: count, dtype: int64

### How many Silver🥈 won by male in top 3 NOC?

In [90]:
df[(df["Medal"] == "Silver") | (df["Gender"] == "Women")]["NOC"].value_counts().head(3)

NOC
USA    1858
URS    1055
GBR     706
Name: count, dtype: int64

### Who owns most GOLD🏅 medal ?

In [110]:
df[df["Medal"] == "Gold"]['Athlete_Name'].value_counts().index[0]

'LATYNINA, Larisa'

### Who owns most GOLD🏅 medal in male and female ?

In [118]:
print(f'{df[(df["Medal"] == "Gold") | (df["Gender"] == "Men")]["Athlete_Name"].value_counts().index[0]} won {df[(df["Medal"] == "Gold") | (df["Gender"] == "Men")]["Athlete_Name"].value_counts().iloc[0]} Golds')

ANDRIANOV, Nikolay won 15 Golds


In [119]:
print(f'{df[(df["Medal"] == "Gold") | (df["Gender"] == "Women")]["Athlete_Name"].value_counts().index[0]} won {df[(df["Medal"] == "Gold") | (df["Gender"] == "Women")]["Athlete_Name"].value_counts().iloc[0]} Golds')

LATYNINA, Larisa won 18 Golds


### Country of most gold owner male and female

In [144]:
Men_most_Gold = df[(df["Medal"] == "Gold") | (df["Gender"] == "Men")]["Athlete_Name"].value_counts().index[0]
Women_most_Gold = df[(df["Medal"] == "Gold") | (df["Gender"] == "Women")]["Athlete_Name"].value_counts().index[0]
df[df['Athlete_Name'].isin([Men_most_Gold,Women_most_Gold])]["NOC"].unique()[0]

'URS'

In [122]:
df[(df["Medal"] == "Gold") | (df["Gender"] == "Women")][["Athlete_Name", "NOC"]]

Athlete_Name  NOC
0           FLAMENG, Léon  FRA
4           BURKE, Thomas  USA
6           HAJOS, Alfred  HUN
9      MALOKINIS, Ioannis  GRE
12           MASSON, Paul  FRA
...                   ...  ...
27169  LOGOUNOVA, Tatiana  RUS
27170       SIVKOVA, Anna  RUS
27171      BOKEL, Claudia  GER
27172     DUPLITZER, Imke  GER
27173   HEIDEMANN, Britta  GER

[13566 rows x 2 columns]

In [145]:
df.head(3)

Year    City      Sport     Discipline        Athlete_Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
2  1896  Athens  Athletics      Athletics       LANE, Francis  USA    Men   

   Event Event_Gender   Medal  Position  
0  100km            M    Gold         1  
1  100km            M  Silver         2  
2   100m            M  Bronze         3

### How many different games/discipline/Event was organized in Olympic 1896 ?

In [155]:
print(f"No of Games : {df['Sport'].value_counts().count()}")
print(f"No of Discipline : {df['Discipline'].value_counts().count()}")
print(f"No of Event : {df['Event'].value_counts().count()}")

No of Games : 42
No of Discipline : 55
No of Event : 572


### How many cities participated in the tournament ?

In [157]:
print(f"No of Cities : {df['City'].value_counts().count()}")

No of Cities : 21


### From which city least player participated ?

In [162]:
print(f"No of Cities : {df['City'].value_counts().tail(3)}")

No of Cities : City
Berlin       875
Amsterdam    710
St Louis     470
Name: count, dtype: int64


### From which city least female player participated ?

In [164]:
print(f"No of Cities : {df[df['Gender'] == 'Female']['City'].value_counts().tail(3)}")

No of Cities : Series([], Name: count, dtype: int64)
